Update spec expectation

NobodysNightmare opened the pull request:
SSO features were moved to the professional plan, but this spec was not updated in the process.

# Ticket
https://community.openproject.org/projects/gmbh/work_packages/63572/activity

NobodysNightmare committed:
Update spec expectation

SSO features were moved to the professional plan, but
this spec was not updated in the process.

oliverguenther commented on modules/auth_saml/spec/features/administration/saml_crud_spec.rb:
```diff
@@ -195,7 +195,7 @@
     it "renders the upsale page" do
       visit "/admin/saml/providers"
-      expect(page).to have_text "Available only through the Premium enterprise plan"
+      expect(page).to have_text "Available only through the Professional enterprise plan"
```
```suggestion
      expect(page).to have_enterprise_banner(:professional)
```

oliverguenther reviewed (changes requested):
Thanks for looking into the fix. There's a helper to test for the banner that you can use instead of matching the text, which will probably change again

NobodysNightmare committed:
Use test helper to check for presence of enterprise banner

This should be more stable, since we can change expectations regarding enterprise banners in a single place.

Co-authored-by: Oliver Günther <o.guenther@openproject.com>

NobodysNightmare commented:
Tests are pretty slow today and then being canceled after 40 minutes -.-

Also happened to me at least once on an unrelated branch.

NobodysNightmare commented:
I am going to self-merge this now, so that we get a successful CI pipeline back. All of your review feedback should have been applied to this PR.
